src/main/java/io/jenkins/plugins/audit/listeners/SaveableChangeListener.java

```diff
@@ -0,0 +1,45 @@
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.Extension;
+import hudson.XmlFile;
+import hudson.model.Fingerprint;
+import hudson.model.Saveable;
+import hudson.model.listeners.SaveableListener;
+import io.jenkins.plugins.audit.event.CredentialsUsage;
+import org.apache.logging.log4j.audit.LogEventFactory;
+
+import java.util.Hashtable;
+
+
+@Extension
+public class SaveableChangeListener extends SaveableListener {
+    /**
+     * Fired when a saveable object is created. But for now this is customized to only log
+     * the saveable fingerprint instances.
+     *
+     * @param o the saveable object.
+     * @param file the XmlFile for this saveable object.
+     */
+    @Override
+    public void onChange(Saveable o, XmlFile file) {
+        CredentialsUsage credentialsUsage = LogEventFactory.getEvent(CredentialsUsage.class);
+        if(o instanceof Fingerprint){
+
+            Fingerprint fp = (Fingerprint) o;
+            credentialsUsage.setFileName(fp.getFileName());
+            credentialsUsage.setName(fp.getDisplayName());
+            credentialsUsage.setTimestamp(fp.getTimestampString());
+            try{
+                fp.getUsages().forEach((k, v) ->{
+                    credentialsUsage.setUsage(v.toString());
+                    credentialsUsage.logEvent();
+                });
+            } catch (NullPointerException npe) {
```

Comment:
You shouldn't catch NullPointerException usually. What here could possibly be null? If `fp.getUsages()` returns null, for instance, you should check its return value for null first before calling forEach on it.

Reply:
I had such checks before. But it fails in the build. Let me check if it still fails.

Reply:
```
Possible null pointer dereference in io.jenkins.plugins.audit.listeners.SaveableChangeListener.onChange(Saveable, XmlFile) due to return value of called method [io.jenkins.plugins.audit.listeners.SaveableChangeListener, io.jenkins.plugins.audit.listeners.SaveableChangeListener] Dereferenced at SaveableChangeListener.java:[line 35]Known null at SaveableChangeListener.java:[line 35] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
```

It still fails with this error and I could solve it only when I was handling the exception. But looks like there are very less chance for that to be null.